src/multivariate/mvtdist.jl

```diff
@@ -72,11 +101,14 @@ IsoTDist(df::Real, d::Int, σ::Real) = GenericMvTDist(df, ScalMat(d, abs2(σ)))
 mvtdist(df::Real, μ::Vector, C::AbstractPDMat) = GenericMvTDist(df, μ, C)
 mvtdist(df::Real, C::AbstractPDMat) = GenericMvTDist(df, C)
 
-mvtdist(df::Real, μ::Vector, σ::Real) = GenericMvTDist(df, μ, ScalMat(length(μ), abs2(σ)))
+mvtdist(df::Real, μ::AbstractVector, σ::Real) = GenericMvTDist(df, μ, ScalMat(length(μ), abs2(σ)))
 mvtdist(df::Real, d::Int, σ::Real) = GenericMvTDist(df, μ, ScalMat(d, abs2(σ)))
-mvtdist(df::Real, μ::Vector, σ::Vector) = GenericMvTDist(df, μ, PDiagMat(abs2.(σ)))
-mvtdist(df::Real, μ::Vector, Σ::Matrix) = GenericMvTDist(df, μ, PDMat(Σ))
-mvtdist(df::Real, Σ::Matrix) = GenericMvTDist(df, PDMat(Σ))
+mvtdist(df::Real, μ::AbstractVector, σ::Vector) = GenericMvTDist(df, μ, PDiagMat(abs2.(σ)))
+mvtdist(df::Real, μ::AbstractVector, Σ::AbstractMatrix) = GenericMvTDist(df, μ, PDMat(Σ))
+mvtdist(df::Real, Σ::AbstractMatrix) = GenericMvTDist(df, PDMat(Σ))
+
+mvtdist(df::Real, μ::AbstractVector, Σ::UniformScaling{<:Real}) = GenericMvTDist(df, μ, ScalMat(length(μ), Σ.λ))
+
```

Comment:
This is not part of the API and should probably be removed completely? I don't think there's any advantage over just using constructors. And we don't want to support vectors or scalars for the scale matrix anyway (similar to the deprecations in `MvNormal` etc.) as it only causes confusion (is it squared or not?).

Reply:
Sorry, can you clarify this comment - are you referring to the `mvtdist`? functions? And they should be removed?

Reply:
Well, not in this PR as it is not about these functions. But in general, yes. We should just follow the same approach as in `MvNormal` and other distributions - we should just call the constructor. And we should not operate with scalars or vectors of scale matrices.

Reply:
Actually here the only reason I worked with these utilities rahter than calling the constructor directly was to handle the densification of the output of `X_A_Xt`. Although the output (for these test cases) is a PDMatable, th X_A_Xt call densifies it. In retrospect, defining: 

```
GenericMvTDist(df::Real, μ::AbstractVector, Σ::AbstractMatrix) = GenericMvTDist(df, μ, PDMat(Σ))
```

is more appropriate here, rather than using these helper functions. I'll follow up with this change and these changes should disappear from the change set. 

But agree with your point - these changes make sense.